base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -230,6 +230,22 @@ where B: BlockchainBackend
             txn.set_horizon_data(kernel_sum, utxo_sum);
             blockchain_db.write(txn)?;
             blockchain_db.store_pruning_horizon(config.pruning_horizon)?;
+        } else if blockchain_db.get_chain_metadata()?.pruning_horizon() == 0 {
```

Comment:
The hash should be the same for pruned or archival nodes, so can't we exclude this check?

Reply:
Removed.